### PR TITLE
[Silabs] DIC integration changes under DIC flag and DIC macro

### PR DESCRIPTION
**Pull request by Thirsrin**
Added :

DIC integration changes under DIC flag(enable_dic) and DIC macro(DIC_ENABLE)

Tested :

all builds success for efr32 and SiWx917

### CODE REVIEW

File: third_party/silabs/efr32_sdk.gni [hunk 1]
```diff
@@ -200,6 +200,9 @@ template("efr32_sdk") {
     }
 
     if (defined(invoker.chip_enable_wifi) && invoker.chip_enable_wifi) {
+      if (enable_dic) {
+        defines += [ "DIC_ENABLE=1" ]
```
Comment (jepenven-silabs) [inline third_party/silabs/efr32_sdk.gni:204]: This seems to duplicates the one present in `examples/platform/silabs/efr32/BUILD.gn`

File: third_party/silabs/SiWx917_sdk.gni [hunk 2]
```diff
@@ -125,6 +125,7 @@ template("efr32_sdk") {
       "PLATFORM_HEADER=\"platform-header.h\"",
       "USE_NVM3=1",
       "RSI_ARM_CM4F",
+      "SIWX_917",
```
Comment (jepenven-silabs) [inline third_party/silabs/SiWx917_sdk.gni:128]: I don't see any `DIC_ENABLE=1` for 917 anywhere ...

File: src/platform/silabs/efr32/efr32-chip-mbedtls-config.h [hunk 3]
```diff
@@ -75,6 +80,10 @@
 #define MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 #define MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED
 #define MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+#ifdef DIC_ENABLE
```
Comment (jepenven-silabs) [inline src/platform/silabs/efr32/efr32-chip-mbedtls-config.h:83]: The defines should be all grouped inside a single #ifdef

File: src/platform/silabs/SiWx917/siwx917-chip-mbedtls-config.h [hunk 4]
```diff
@@ -114,8 +126,12 @@ typedef void mbedtls_ecp_restart_ctx;
 #define MBEDTLS_X509_CSR_PARSE_C
 #define MBEDTLS_X509_USE_C
 
-#define MBEDTLS_MPI_WINDOW_SIZE 1       /**< Maximum windows size used. */
-#define MBEDTLS_MPI_MAX_SIZE 32         /**< Maximum number of bytes for usable MPIs. */
+#define MBEDTLS_MPI_WINDOW_SIZE 1 /**< Maximum windows size used. */
+#ifdef DIC_ENABLE
```
Comment (jepenven-silabs) [inline src/platform/silabs/SiWx917/siwx917-chip-mbedtls-config.h:130]: Same comments these should all be grouped together

File: src/lwip/silabs/lwipopts-wf200.h [hunk 5]
```diff
@@ -40,6 +40,9 @@
 #define MEM_LIBC_MALLOC (1)
 #define mem_clib_free vPortFree
 #define mem_clib_malloc pvPortMalloc
+#ifdef DIC_ENABLE
```
Comment (jepenven-silabs) [inline src/lwip/silabs/lwipopts-wf200.h:43]: Same comment please group all defines together under a single `#ifdef`

File: examples/platform/silabs/efr32/BUILD.gn [hunk 6]
```diff
@@ -234,6 +261,9 @@ config("silabs-wifi-config") {
     # TODO: Change this macro once WF200 support is provided
     defines += [ "WIFI_ENABLE_SECURITY_WPA3=1" ]
   }
+  if (enable_dic) {
+    defines += [ "DIC_ENABLE=1" ]
```
Comment (jepenven-silabs) [inline examples/platform/silabs/efr32/BUILD.gn:265]: This seems to be duplicated with the one in `efr32_sdk.gni`

File: examples/platform/silabs/SiWx917/BUILD.gn [hunk 7]
```diff
@@ -42,6 +42,33 @@ config("chip_examples_project_config") {
...
+      "${chip_root}/examples/platform/silabs/DIC/matter_abs_interface/src/dic.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/MQTT_transport.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/altcp.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/altcp_alloc.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/altcp_tcp.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/altcp_tls_mbedtls.c",
+      "${chip_root}/examples/platform/silabs/DIC/mqtt_transport_interface/src/altcp_tls_mbedtls_mem.c",
+    ]
+    public_deps = [
+      "${chip_root}/src/inet",
```
Comment (jepenven-silabs) [inline examples/platform/silabs/SiWx917/BUILD.gn:65]: I don't seen any `DIC_ENABLE=1` for 917

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]

**Review by jepenven-silabs — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**